### PR TITLE
Homework10: Mariya Simonenko

### PR DESCRIPTION
**Pull request by Mary-si**

### CODE REVIEW

**Comment by Stml89** [comment]
Hello @Mary-si , the pylint fails PR because of space in the filename:
image

**Comment by Mary-si** [comment]
спасибо((((

File: homework10_caching_func.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+@cache
+def fibonacci(n):
+    """выполнение """
+    if n <= 1:
+        return n
+    return fibonacci(n - 1) + fibonacci(n - 2)
+
+
+# Проверка
+print(fibonacci(5))  # Вывод: 5
```
Comment (Stml89) [inline homework10_caching_func.py:57]: Use `assert` to check your code

File: homework10_type_decorator.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    print(a + b)
+
+
+# Проверка
+add("3", 5)
+add(5, 5)
+add('a', 'b')
+
+
+def typed1(expected_type):
```
Comment (Stml89) [inline homework10_type_decorator.py:52]: it should be 1 decorator that supports different amount of arguments

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: homework10_caching_func.py (hunk 1 above)
Comment (Mary-si) [reply homework10_caching_func.py:57]: поправила

File: homework10_type_decorator.py (hunk 2 above)
Comment (Mary-si) [reply homework10_type_decorator.py:52]: поправила

**Pull request merged**